implementation.js

```diff
@@ -78,7 +78,7 @@ if (hasSymbols) {
 		iteratorSymbol = '@@iterator';
 	} else {
 		try {
-			if (Function('for (var x of [0]) x;')() === 0) { // eslint-disable-line no-new-func
+			if (Function('var s = new Set(); s.add(0); for (var x of s) return x;')() === 0) { // eslint-disable-line no-new-func
```

Comment:
I think this is fine - but I think it would be ideal to only have one `Function` occurrence in the file - ie, to use the definition on line 82 both for `forOf` and for the `if` check.